### PR TITLE
Adding ability to install benthos with `asdf`

### PR DESCRIPTION
**Pull request by rolandtritsch**
This PR addresses/implements issue #1568 - https://github.com/benthosdev/benthos/issues/1568

### CODE REVIEW

**Comment by rolandtritsch** [comment]
Hi @mihaitodor. This is ready for review. You can test drive it with ...

```
asdf plugin add benthos https://github.com/rolandtritsch/benthos-asdf.git
```

Feedback welcome.

File: bin/install.sh [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
```
Comment (rolandtritsch) [inline bin/install.sh]: This is temporary. `install.sh` can be removed, when this PR is merged - https://github.com/benthosdev/benthos/pull/1614

Comment (mihaitodor) [reply bin/install.sh]: Thanks @rolandtritsch! I'll let Ash merge that PR, so we don't have to duplicate the code here.

File: README.md [hunk 2]
```diff
@@ -47,15 +27,77 @@ asdf global benthos latest
...
+benthos         4.10.0          /home/roland/.tool-versions
+```
+
 # Contributing
 
-Contributions of any kind welcome! See the [contributing guide](contributing.md).
+Contributions of any kind welcome! See the [contributing][] guide.
 
-[Thanks goes to these contributors](https://github.com/benthosdev/asdf-benthos/graphs/contributors)!
+Thanks goes to these [contributors][]!
```
Comment (mihaitodor) [inline README.md:91]: Guess this should point to `CONTRIBUTING.md`

LE: It's fine. I didn't notice it's pointing to a named anchor.

**Review by mihaitodor — COMMENTED** [review]
Nice one, I really appreciate the work you put into this and thanks for the patience! I saw a small issue in the readme, but I'm happy to merge it once that's fixed and it should all work when https://github.com/benthosdev/benthos/pull/1614 gets released.

**Comment by rolandtritsch** [comment]
Hi @mihaitodor. Made the GH workflows work.

image

But ... for it to work I had to configure a GH_API_TOKEN.

image

The good news is it is a token with very limited scope (`public_repo`).

I also change the code to use the install script from the repo directly. 

Right now I would suggest to merge the PR as it is (Note: You need to create `secret.GH_API_TOKEN` first). 

I will continue to iterate on the plugin with smaller PRs as needed.

WDYT?

**Comment by mihaitodor** [comment]
Sweet, thanks @rolandtritsch! I think it works. I followed the instructions from [here](https://github.com/asdf-vm/asdf/blob/master/docs/plugins/create.md#github-api-rate-limiting) to create a token.

**Pull request merged**